robot/client/client_session_test.go

```diff
@@ -267,68 +288,85 @@ func TestClientSessionOptions(t *testing.T) {
+// Test that once a session has expired, the next call will start a new session.
 func TestClientSessionExpiration(t *testing.T) {
 	t.Parallel()
 	ctx := context.Background()
-	for _, webrtcDisabled := range []bool{false, true} {
-		webrtcDisabledCopy := webrtcDisabled
+	for _, webrtcDisabledOuter := range []bool{false, true} {
+		webrtcDisabled := webrtcDisabledOuter
 
 		t.Run(
 			fmt.Sprintf(
 				"webrtc disabled=%t",
-				webrtcDisabledCopy,
+				webrtcDisabled,
 			),
 			func(t *testing.T) {
 				t.Parallel()
 
 				logger := logging.NewTestLogger(t)
 
 				sessMgr := &sessionManager{}
-				arbName := resource.NewName(echoAPI, "woo")
 
-				var dummyEcho1 dummyEcho
 				injectRobot := &inject.Robot{
-					ResourceNamesFunc: func() []resource.Name { return []resource.Name{arbName} },
-					ResourceByNameFunc: func(name resource.Name) (resource.Resource, error) {
-						return &dummyEcho1, nil
-					},
-					MachineStatusFunc: func(_ context.Context) (robot.MachineStatus, error) {
+					MachineStatusFunc: func(ctx context.Context) (robot.MachineStatus, error) {
 						return robot.MachineStatus{State: robot.StateRunning}, nil
 					},
 					ResourceRPCAPIsFunc: func() []resource.RPCAPI { return nil },
 					LoggerFunc:          func() logging.Logger { return logger },
 					SessMgr:             sessMgr,
 				}
 
+				// Inject a base. Its actuating methods- SetPower, MoveStraight, etc. will start sessions.
+				injectBase := &inject.Base{}
+				rs := map[resource.Name]resource.Resource{
+					base.Named("base1"): injectBase,
+				}
+				injectRobot.MockResourcesFromMap(rs)
+
 				svc := web.New(injectRobot, logger)
 
 				options, _, addr := robottestutils.CreateBaseOptionsAndListener(t)
 				err := svc.Start(ctx, options)
 				test.That(t, err, test.ShouldBeNil)
 
 				var opts []client.RobotClientOption
-				if webrtcDisabledCopy {
+				if webrtcDisabled {
 					opts = append(opts, client.WithDialOptions(rpc.WithWebRTCOptions(rpc.DialWebRTCOptions{
 						Disable: true,
 					})))
 				}
 
-				injectRobot.Mu.Lock()
 				var capSessID uuid.UUID
-				injectRobot.MachineStatusFunc = func(ctx context.Context) (robot.MachineStatus, error) {
+				injectRobot.Mu.Lock()
+				// Repurpose to capture Session ID
+				injectBase.MoveStraightFunc = func(ctx context.Context, distanceMm int, mmPerSec float64, extra map[string]interface{}) error {
 					sess, ok := session.FromContext(ctx)
 					if !ok {
 						panic("expected session")
 					}
 					capSessID = sess.ID()
-					return robot.MachineStatus{State: robot.StateRunning}, nil
+					return nil
+				}
+				injectBase.SetPowerFunc = func(ctx context.Context, linear, angular r3.Vector, extra map[string]interface{}) error {
+					sess, ok := session.FromContext(ctx)
+					test.That(t, ok, test.ShouldBeTrue)
+					if !ok {
+						panic("expected session")
+					}
+					capSessID = sess.ID()
+					// associate with this context. Unused?
+					session.SafetyMonitorResourceName(ctx, someTargetName2)
```

Comment:
if we don't use this/check the context lets remove. In theory if a resource is safety monitored, we should get a Stop() when session expires. But since we mock the session manager, I don't think we can check that in this test